File: eclipse-plugin/com.atec.sourcetrace.eclipse/src/com/atec/sourcetrace/eclipse/core/ApiPaths.java
package com.atec.sourcetrace.eclipse.core;

/**
 * Backend v2.6 API paths — must stay identical to vscode-extension
 * {@code serverConfig.ts} PATHS. Do not invent Eclipse-only endpoints.
 */
public final class ApiPaths {

	public static final String HEALTH = "/api/health";
	public static final String EQUIPMENT_LIST = "/api/equipment";
	public static final String TRACE_REPORT = "/api/trace/report";
	public static final String TRACE_SELECTION = "/api/trace/selection";

	private ApiPaths() {
	}

	public static String equipmentById(int id) {
		return EQUIPMENT_LIST + "/" + id;
	}

	public static String equipmentRepositories(int id) {
		return EQUIPMENT_LIST + "/" + id + "/repositories";
	}
}

File: visualstudio-extension/src/Atec.SourceTrace.Core/ApiPaths.cs
namespace Atec.SourceTrace.Core;

/// <summary>Backend v2.6 API paths — identical to vscode-extension serverConfig PATHS.</summary>
public static class ApiPaths
{
    public const string Health = "/api/health";
    public const string EquipmentList = "/api/equipment";
    public const string TraceReport = "/api/trace/report";
    public const string TraceSelection = "/api/trace/selection";

    public static string EquipmentById(int id) => $"{EquipmentList}/{id}";

    public static string EquipmentRepositories(int id) => $"{EquipmentList}/{id}/repositories";
}
